enhancements/20220525-system-logging.md

```diff
@@ -0,0 +1,89 @@
+# System Logging
+
+## Summary
+
+We need to be able to support exporting harvester system logs outside the cluster, to allow for collecting and analyzing those logs to aid in debugging and harvester management. 
+
+### Related Issues
+
+- https://github.com/harvester/harvester/issues/577
+
+## Motivation
+
+### Goals
+
+List the specific goals of the enhancement. How will we know that this has succeeded?
+
+- The user can aggregate harvester logs in a centralized location
+- The user can view the aggregated harvester logs outside the cluster
+
+### Non-goals
+
+- this enhancement does not cover integration into the harvester / rancher UI (but ideally this will eventually be implemented as well)
+
+## Proposal
+
+We will deploy a new `ManagedChart` in the [Harvester Installer](https://github.com/harvester/harvester-installer) to
+install. The `ManagedChart` will deploy a [`ClusterFlow`](https://banzaicloud.com/docs/one-eye/logging-operator/configuration/flow/) to select and 
+aggregate interesting logs, which can be managed by a new [`ClusterOutput`](https://banzaicloud.com/docs/one-eye/logging-operator/configuration/output/) 
+crd. The `ClusterOutput` can then be configured by settings from the harvester UI.
+
+### User Stories
+
+Currently, users need to manually check harvester for failing pods or services and manually check logs using `kubectl` or other cluster
+inspection tools.
+
+#### Setup
+
+Logging will be available after the cluster is installed, or upgraded from a previous version.
+
+#### Configuration
+
+The logging behavior should be configuratble using the Harvester UI settings.
+
+##### Outputs
+
+There are a lot of supported [plugins](https://banzaicloud.com/docs/one-eye/logging-operator/configuration/plugins/) so we probably want 
+to pick a few "supported" pluggins ([splunk](https://banzaicloud.com/docs/one-eye/logging-operator/configuration/plugins/outputs
+/splunk_hec/), [Elasticsearch](https://banzaicloud.com/docs/one-eye/logging-operator/configuration/plugins/outputs/elasticsearch/), etc) 
+to wrap in a clean UI (similar to VM provisioning), and allow users to manually enter yaml for greater control over the `ClusterOutput`.
+
+Banzai does not impose a limit on the amount of outputs, so barringto performance issues we should not need to enforce limits on the amount of outputs a user can configure.
+
+##### Log Level
+
+We can add a Flow [filter](https://banzaicloud.com/docs/one-eye/logging-operator/configuration/plugins/filters/) to filter logs based on 
+entry level. When the love level changes we will need to path the current `ClusterFlow`.
+
+#### Viewing Harvester Logs
```

Comment:
Per https://kubernetes.io/docs/concepts/cluster-administration/logging/
```
In a cluster, logs should have a separate storage and lifecycle independent of nodes, pods, or containers. This concept is called cluster-level logging.

Cluster-level logging architectures require a separate backend to store, analyze, and query logs. Kubernetes does not provide a native storage solution for log data. 
```

For Harvester, the feasible way is to use LH as the cluster level logging storage. A bit similar to `promethues`, which uses a LH PVC to store related data.